dialogflow-cx/streaming_detect_intent_infinite.py

```diff
@@ -0,0 +1,720 @@
+
+                combined_data: bytes = b"".join(data)
+                yield combined_data
+
+            except asyncio.TimeoutError:
+                logger.debug(
+                    "[generator] No audio chunk received within timeout, continuing..."
+                )
+                continue
+
+    def _extract_wav_header(self, audio_data: bytes) -> bytes:
+        """Extracts the WAV header from the audio data.
+
+        Args:
+            audio_data: The audio data potentially containing a WAV header.
+
+        Returns:
+            The audio data without the WAV header.
+        """
+        if audio_data[:4] == b"RIFF":
+            try:
+                header: Tuple[
+                    bytes, int, bytes, bytes, int, int, int, int, int, int, bytes, int
+                ] = struct.unpack("<4sI4s4sIHHIIHH4sI", audio_data[:44])
+                logger.debug(f"WAV header detected: {header}")
+                return audio_data[44:]
+            except struct.error as e:
+                logger.error(f"Error unpacking WAV header: {e}")
+                return audio_data
+        else:
+            logger.warning("No WAV header found.")
+            return audio_data
+
+    def play_audio(self, audio: bytes) -> None:
+        """Plays audio from the given bytes data and removes WAV header."""
+        # Remove WAV header if present
+        audio = self._extract_wav_header(audio)
+
+        # Play the raw PCM audio
+        try:
+            self._output_audio_stream.start_stream()
+            self._output_audio_stream.write(audio)
+        finally:
+            self._output_audio_stream.stop_stream()
+
+
+class DialogflowCXStreaming:
+    """Manages the interaction with the Dialogflow CX Streaming API."""
+
+    def __init__(
+        self,
+        agent_name: str,
+        language_code: str,
+        model: str,
+        dialogflow_timeout: float,
+        debug: bool = False,
+        sample_rate: int = 16000,
+        audio_encoding: str = "LINEAR16",
+        output_audio_encoding: str = "LINEAR16",
+        voice: Optional[str] = None,
+        single_utterance: bool = False,
+    ) -> None:
+        """Initializes the Dialogflow CX Streaming API client."""
+        try:
+            _, project, _, location, _, agent_id = agent_name.split('/')
+        except ValueError:
+            raise ValueError('Invalid agent name format. Expected format: projects/<project>/locations/<location>/agents/<agent_id>')
+        if location != 'global':
+            client_options: ClientOptions = ClientOptions(
+                api_endpoint=f'{location}-dialogflow.googleapis.com',
+                quota_project_id=project,
+            )
+        else:
+            client_options = ClientOptions(quota_project_id=project)
+        self.client: dialogflowcx_v3.SessionsAsyncClient = (
+            dialogflowcx_v3.SessionsAsyncClient(client_options=client_options)
+        )
+        self.agent_name: str = agent_name
```

Comment:
Same here, please only keep whatever is necessary to keep state. Configurations should be inlined.